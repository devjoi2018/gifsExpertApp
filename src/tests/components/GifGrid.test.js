import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import {GifGrid} from '../../../src/components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

// Finge cualquier llamada que haga el archivo useFetchGifs y controla
// la informacion que el metodo devuelve
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'Categoria de prueba';

    test('Debe de mostrar el componente de forma correcta', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de mostrar items cuando se cargan imagenes en useFetchGifs', () => {

        const gifs = [{
            id: 1,
            url: "http://url.com/test.jpg",
            title: "Titulo de la imagen",
          }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });
    
});
