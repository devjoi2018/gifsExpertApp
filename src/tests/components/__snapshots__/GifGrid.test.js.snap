// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar el componente de forma correcta 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Categoria de prueba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan imagenes en useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Categoria de prueba
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Titulo de la imagen"
      url="http://url.com/test.jpg"
    />
  </div>
</Fragment>
`;
